### PR TITLE
tests: take screenshots of (almost) all pages from the source & journalist interface

### PR DESCRIPTION
**Pull request by dachary**
## Status

Ready for review

## Description of Changes

Contributes-to-the-resolution-of #2016

Take screenshots of (almost) all pages from the source & journalist interface. The logic to navigate to a given state is shared with functional tests but the purpose is human verification of the page layout rather than automated validation of the interaction steps.

In addition to providing an automated way to create screenshots for guides, it also helps catch typos embedded in code run via the jinja templates. This is specially important when the templates are modified to include calls to gettext functions. It will also help verify l10n in a given language looks ok.

## Testing

* pytest -v tests/pages-layout
* geeqie tests/pages-layout/screenshots/en_US

## Deployment

This is happening during tests only, no impact on deployment.

### If you made changes to the app code:

- [x] Unit and functional tests pass on the development VM

### CODE REVIEW

**Comment by dachary** [comment]
rebased & repushed

File: securedrop/tests/pages-layout/functional_test.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        self.driver.quit()
+
+    def _screenshot(self, filename):
+        self.driver.save_screenshot(os.path.join(self.log_dir, filename))
+
+    def _javascript_toggle(self):
+        # the following is a noop for some reason, workaround it
+        # profile.set_preference("javascript.enabled", False)
+        # https://stackoverflow.com/a/36782979/837471
```
Comment (redshiftzero) [inline securedrop/tests/pages-layout/functional_test.py]: Hm, weird. Well this certainly works (as evidenced by `source-index_javascript` having the warning bar about using Firefox that `source-index` does not) so a fine workaround for now 👌

**Comment by dachary** [comment]
> the tests take significantly longer

Right. And it will get worse as we add translations. I'll add an environment variable to conditionally enable them.

**Comment by dachary** [comment]
> This also gets us so close to completion of the time-saving ticket #2016

Is there a way to say **Contributes-to-the-resolution-of: #2016** instead of **Fixes: #2016** ?

**Comment by dachary** [comment]
> I'll add an environment variable to conditionally enable them.

Strike that. I wrongly assumed #2164 was another PR. I cherry-picked the commit from #2164, it's a really great way to do it. I did not know pytest had that kind of flexibility :-) 💯

**Comment by heartsucker** [comment]
>  I don't think we want them to be executed every time that a developer is running `pytests -v tests/`

It screws up the traceback too.

**Comment by dachary** [comment]
> It screws up the traceback too.

@heartsucker how so ?

**Comment by heartsucker** [comment]
```
args = (<test_journalist.TestJournalistLayout instance at 0x7f6f3c73d248>, 'dellsberg', 'pentagonpapers'), kwargs = {'hotp': None, 'is_admin': False}
curframe = <frame object at 0x2845300>
calframe = [(<frame object at 0x2845300>, '/vagrant/securedrop/tests/functional/step_helpers.py', 15, 'wrapper', ['        curfra...ytest/vendored_packages/pluggy.py', 265, '__init__', ['        try:\n', '            self.result = func()\n'], 1), ...]
locals = {'add_user_btn': <selenium.webdriver.remote.webelement.WebElement (session="e303db5a-78b5-4b93-94c4-d4211e4d98bf", ele...68-4ecc-866b-f04bb04b6c47}")>, 'hotp': None, 'self': <test_journalist.TestJournalistLayout instance at 0x7f6f3c73d248>}
class_name = 'TestJournalistLayout'
x = ('/vagrant/securedrop/tests/functional/step_helpers.py', 24, 'wrapper', 'stack = [x for x in traceback.extract_stack()')
stack = [('/vagrant/securedrop/tests/functional/journalist_navigation_steps.py', 537, '_admin_creates_a_user', 'hotp=hotp)'), ('/vagrant/securedrop/tests/functional/step_helpers.py', 24, 'wrapper', 'stack = [x for x in traceback.extract_stack()')]
path = 'journalist_navigation_steps.py-TestJournalistLayout-_admin_creates_a_user'

    def wrapper(*args, **kwargs):
        curframe = inspect.currentframe()
        calframe = inspect.getouterframes(curframe, 2)
    
        locals = calframe[1][0].f_locals
        if "testfunction" in locals:
            fun = calframe[1][0].f_locals["testfunction"]
            class_name = fun.__self__.__class__.__name__
        else:
            class_name = calframe[1][0].f_locals["self"].__class__.__name__
    
        stack = [x for x in traceback.extract_stack()
                 if '/tests/functional' in x[0]]
        path = ('-'.join([stack[0][0].split('/')[-1], class_name] +
                         [x[2] for x in stack if x[2] is not 'wrapper']))
        if screenshots_enabled:
            image_path = join(LOG_DIR, '%s-before.png' % path)
            args[0].driver.save_screenshot(image_path)
>       result = f(*args, **kwargs)
E       TypeError: _add_user() got multiple values for keyword argument 'is_admin'

tests/functional/step_helpers.py:31: TypeError
```

This is a less useful traceback than what `pytest` normally would display.

**Comment by redshiftzero** [comment]
Seems like we should disable the `@screenshots` decorator now that we have the `--page-layout` option. We can run that in CI (and we can add those screenshots as test artifacts)

**Comment by dachary** [comment]
> tests/functional/step_helpers.py:31: TypeError

You had me worried :-) This is bad indeed but it does not come from this pull request. 

I should have forseen that the clever frame inspection hack would interfere with the clever pytest error reporting. Completely misse that when reviewing the `@screenshots` pull request.

**Pull request merged**